evaluator.js

```diff
@@ -128,7 +126,7 @@ function analyze_constant_declaration(stmt) {
     return (env, succeed, fail) => {
         value_func(env,
                    (value, fail2) => {
-                    set_name_value(name, value, env);
+                    set_name_value(name, value, env, false);
```

Comment:
The `false` argument is necessary for `set_name_value` as it indicates that the name being declared is not a variable.